### DISCUSSION TITLE
`npm outdated` should list deprecated packages as well

### DISCUSSION THREAD

**Discussion by k2snowman69** [npm/rfcs#670] [RRFC]
### Problem
Currently there isn't a great way to easily visualize all packages that are **deprecated**.

Historically you could run `npm install` and it would output deprecated packages during/after the install. However with npm@8.19.2 I cannot find an easy way to list all deprecated packages within a repository currently:

 - `npm i --ignore-scripts --no-fund --no-audit` - Only outputs the added packages no deprecation warnings
 - `npm i --ignore-scripts --no-fund` - Outputs audit info but no deprecation warnings
 - `npm i` - Still no deprecation warnings

Further `npm outdated` doesn't include deprecations or deprecation messages to help keep your packages up to date. IMO a deprecated package is something that could classify as "outdated".

With that said, IMO historically depending on `npm i` to list deprecated packages in previous versions of npm wasn't ideal to begin with.

### Proposal

Add deprecated as a column for `npm outdated`. The way the output could look like is the following:

| Package                   | Current | Wanted  | Latest  | Location                               | Depended by  | Deprecation message                                       |
| ------------------------- | ------- | ------- | ------- | -------------------------------------- | ------------ | --------------------------------------------------------- |
| @snowcoders/sortier       | 3.3.0   | 3.3.0   | 3.3.0   | node_modules/@snowcoders/sortier       | some-package | Renamed to just sortier (no organization). npm i sortier. |
| @types/js-base64          | 2.3.2   | 2.3.2   | 3.3.1   | node_modules/@types/js-base64          | some-package |                                                           |
| @types/react              | 18.0.26 | 18.0.26 | 18.0.27 | node_modules/@types/react              | some-package |                                                           |
| deep-equal                | 2.0.5   | 2.2.0   | 2.2.0   | node_modules/deep-equal                | some-package |                                                           |
| html-react-parser         | 3.0.4   | 3.0.8   | 3.0.8   | node_modules/html-react-parser         | some-package |                                                           |
| informed                  | 4.35.0  | 4.39.3  | 4.39.3  | node_modules/informed                  | some-package |                                                           |
| jest                      | 28.1.3  | 28.1.3  | 29.4.0  | node_modules/jest                      | some-package |                                                           |
| jest-environment-jsdom    | 28.1.3  | 28.1.3  | 29.4.0  | node_modules/jest-environment-jsdom    | some-package |                                                           |
| js-base64                 | 2.6.4   | 2.6.4   | 3.7.4   | node_modules/js-base64                 | some-package |                                                           |
| postcss-cli               | 9.1.0   | 9.1.0   | 10.1.0  | node_modules/postcss-cli               | some-package |                                                           |
| resolve-typescript-plugin | 1.2.0   | 1.2.0   | 2.0.0   | node_modules/resolve-typescript-plugin | some-package |                                                           |
| ts-jest                   | 28.0.8  | 28.0.8  | 29.0.5  | node_modules/ts-jest                   | some-package |                                                           |

**Comment by Raymond-Ly** [comment]
If you use NPM V7 or later, you should have a `package-lock.json` with `lockfileVersion: 2`. This lockfile contains the deprecation warnings of all installed packages.

Could also use `npm outdated --depth=5` to check for deeper sub dependencies.

**Reply by ruyadorno** [reply]
starting with npm7, there's also support to an **all** flag to include sub deps, e.g: `npm outdated --all`

**Comment by k2snowman69** [comment]
I think the issue for me was that `lockFileVersion: 2` was missing the deprecation warnings though I'm unsure why not. When I forced the upgrade to `lockFileVersion: 3` on the most recent version of npm the deprecation warnings showed back up in the package-lock.json.

With them being there, it's a simple query to generate the table above so I'm going to close this ticket out.

If I get time I'll try to dig into why I was missing deprecation warnings for my  `lockFileVersion: 2` and open a bug ticket.
